File: test_data/code/java/1122-relative-sort-array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < arr1.length; i++) {
            map.put(arr1[i], map.getOrDefault(arr1[i], 0) + 1);
        }
        List<Integer> list = new ArrayList<>();
        for (int num : arr2) {
            while (map.get(num) > 0) {
                list.add(num);
                map.put(num, map.get(num) - 1);
            }
        }
        for (int num : map.keySet()) {
            if (num < arr2[arr2.length - 1]) {
                while (map.get(num) > 0) {
                    list.add(num);
                    map.put(num, map.get(num) - 1);
                }
            } else {
                Collections.sort(map.keySet().stream().filter(n -> n >= arr2[arr2.length - 1]).collect(Collectors.toList()));
                for (int n : map.keySet().stream().filter(n -> n >= arr2[arr2.length - 1]).collect(Collectors.toList())) {
                    while (map.get(n) > 0) {
                        list.add(n);
                        map.put(n, map.get(n) - 1);
                    }
                }
            }
        }
        int[] res = new int[arr1.length];
        for (int i = 0; i < res.length; i++) {
            res[i] = list.get(i);
        }
        return res;
    }
}

File: test_data/code/python/1122-relative-sort-array.py
Python
from typing import List

class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        count_map = {}
        for num in arr1:
            if num in count_map:
                count_map[num] += 1
            else:
                count_map[num] = 1
        
        sorted_output = []
        for num in arr2:
            if num in count_map:
                sorted_output.extend([num] * count_map[num])
                del count_map[num]
        
        for num in sorted(count_map.keys()):
            sorted_output.extend([num] * count_map[num])
        
        return sorted_output